test/others/crit/test.sh

```diff
@@ -16,8 +16,7 @@ function _exit {
 }
 
 function gen_imgs {
-	setsid ./loop.sh < /dev/null &> /dev/null &
-	PID=$!
+	PID=$( (setsid ./loop.sh < /dev/null &> /dev/null) & jobs -r -p)
```

Comment:
I don't understand how this fixes the race. Could you explain and add a comment to the code? Thanks.

Reply:
I will add a comment